Packet++/src/HttpLayer.cpp

```diff
@@ -890,30 +719,51 @@ void HttpResponseFirstLine::setVersion(HttpVersion newVersion)
 	m_Version = newVersion;
 }
 
-HttpResponseLayer::HttpResponseStatusCode HttpResponseFirstLine::parseStatusCode(const char* data, size_t dataLen)
+HttpResponseStatusCode HttpResponseFirstLine::parseStatusCode(const char* data, size_t dataLen)
 {
 	// minimum data should be 12B long: "HTTP/x.y XXX"
 	if (!data || dataLen < 12)
 	{
-		return HttpResponseLayer::HttpStatusCodeUnknown;
+		return HttpResponseStatusCode::HttpStatusCodeError;
 	}
 
-	const char* statusCodeData = data + 9;
+	std::string statusCodeDataString(data + 9, 3);
 
-	auto codeAsEnum = StatusCodeStringToEnumMap.find(std::string(statusCodeData, 3));
-	if (codeAsEnum == StatusCodeStringToEnumMap.end())
+	for(const auto& pair : statusCodeExplanationStringMap) {
+		if(int(pair.first) == std::stoi(statusCodeDataString)) {
+			return pair.first;
+		}
+	}
+
+	switch(statusCodeDataString[0]) {
+	case '1':{
+		return HttpResponseStatusCode::HttpStatus1xxCodeUnknown;
+	}
+	case '2':{
+		return HttpResponseStatusCode::HttpStatus2xxCodeUnknown;
+	}
+	case '3':{
+		return HttpResponseStatusCode::HttpStatus3xxCodeUnknown;
+	}
+	case '4':{
+		return HttpResponseStatusCode::HttpStatus4xxCodeUnknown;
+	}
+	case '5':{
+		return HttpResponseStatusCode::HttpStatus5xxCodeUnknown;
+	}
+	default:
 	{
-		return HttpResponseLayer::HttpStatusCodeUnknown;
+		return HttpResponseStatusCode::HttpStatusCodeError;
```

Comment:
In all these cases it is better to keep `HttpStatusCodeUnknown` than `HttpStatusCodeError`

Reply:
sure